src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java

```diff
@@ -196,8 +197,9 @@ public HttpResponse doCreateDomain(StaplerRequest req) throws ServletException,
         getStore().checkPermission(MANAGE_DOMAINS);
         JSONObject data = req.getSubmittedForm();
         Domain domain = req.bindJSON(Domain.class, data);
-        if (getStore().addDomain(domain)) {
-            return HttpResponses.redirectTo("./domain/" + Util.rawEncode(domain.getName()));
+        String domainName = domain.getName();
+        if (domainName != null && getStore().addDomain(domain)) {
```

Comment:
Oh wow

I was just pointing out that the conditional logic changed not only the raw
encode stuff.

Yay fixing potential NPEs

On Tue, Apr 19, 2016 at 3:37 AM, Armando Fernández <notifications@github.com
> wrote:

> In
> src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java
> <https://github.com/jenkinsci/credentials-plugin/pull/44#discussion_r60208313>
> :
>
> > @@ -196,8 +197,9 @@ public HttpResponse doCreateDomain(StaplerRequest req) throws ServletException,
> >          getStore().checkPermission(MANAGE_DOMAINS);
> >          JSONObject data = req.getSubmittedForm();
> >          Domain domain = req.bindJSON(Domain.class, data);
> > -        if (getStore().addDomain(domain)) {
> > -            return HttpResponses.redirectTo("./domain/" + Util.rawEncode(domain.getName()));
> > +        String domainName = domain.getName();
> > +        if (domainName != null && getStore().addDomain(domain)) {
>
> @halkeye <https://github.com/halkeye> filed #47
> <https://github.com/jenkinsci/credentials-plugin/pull/47> to address the
> front-end part too. Let's see what the community says.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/credentials-plugin/pull/44/files/98945c79f60bd680a34c56f0b6caab7665e9134c#r60208313>
>



-- 


*Gavin MoganSoftware Developer*

Sauce Labs <http://saucelabs.com/> | Sauce blog <http://sauceio.com/>

Talk to us anytime:  <https://twitter.com/saucelabs>
<https://www.linkedin.com/company/sauce-labs>
<https://plus.google.com/+SauceLabs_Official/posts>
<https://www.facebook.com/saucelabs>